### PR TITLE
Add SnapLayout Options support for Win11

### PR DESCRIPTION
**Pull request by SlimeNull**
Add SnapLayout Options support for Win11

image

### CODE REVIEW

**Comment by ElektroKill** [comment]
Hello,

Sorry for the late response.

I like the idea of adding the support for the Win 11 snap layout feature, however, I'd prefer if this was directly integrated into dnSpy code base. I do not want to depend on additional WPF dependencies.

If this can be done cleanly and cleanly integrated into dnSpy codebase, I'm all for such a change. Please let me know if you are willing to undergo this change.

**Comment by SlimeNull** [comment]
OK, I'll copy the code directly into the dnSpy repository so that it doesn't have to reference the library.

File: dnSpy/dnSpy/Controls/WinSysButton.cs [hunk 1]
```diff
@@ -67,6 +70,33 @@ sealed class WinSysButton : TabButton {
 				window.StateChanged += window_StateChanged;
 		}
 
+		void WinSysButton_Click(object sender, RoutedEventArgs e) {
+
```
Comment (ElektroKill) [inline dnSpy/dnSpy/Controls/WinSysButton.cs:74]: Empty new line after opening brace, please remove it.

File: dnSpy/dnSpy/Lib/EleCho.WpfSuite/Utilities/WindowOption.NativeMethods.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+				TEXT_COLOR,
+				VISIBLE_FRAME_BORDER_THICKNESS,
+				SYSTEMBACKDROP_TYPE,
+				LAST
+			}
+
+			// Token: 0x0200002E RID: 46
+			[Flags]
+			public enum SetWindowPosFlags {
+				// Token: 0x04000368 RID: 872
```
Comment (ElektroKill) [inline dnSpy/dnSpy/Lib/EleCho.WpfSuite/Utilities/WindowOption.NativeMethods.cs:145]: Remove these comments which are decompiler artifact.

**Review by ElektroKill — CHANGES_REQUESTED** [review]
Please correct the code style issues I mentioned in the review, furthermore, would it be possible to isolate the code required for the Snap layouts features and remove the code for other features like rounded corners (already implemented in dnSpy) and others like backdrop, caption styling, etc? preferably the PR should include only the code necessary for the snaplayouts feature. It will make the code much easier to read as it will only contain the necessary logic!

Let me know what you think of these suggestions.

**Comment by SlimeNull** [comment]
Fixed : )

**Comment by SlimeNull** [comment]
If there's anything else you'd like to change, please let me know.

**Review by ElektroKill — APPROVED** [review]

**Comment by ElektroKill** [comment]
Thanks!

**Pull request merged**